homeassistant/components/venstar/sensor.py

```diff
@@ -0,0 +1,138 @@
+"""Representation of Venstar sensors."""
+
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+
+from . import VenstarDataUpdateCoordinator, VenstarEntity
+from .const import DOMAIN
+
+
+@dataclass
+class VenstarSensorEntityDescription(SensorEntityDescription):
+    """Base description of a Sensor entity."""
+
+
+HUMIDITY_SENSOR_DESCRIPTION = VenstarSensorEntityDescription(
+    key="humidity",
+    device_class=DEVICE_CLASS_HUMIDITY,
+    state_class=STATE_CLASS_MEASUREMENT,
+    native_unit_of_measurement=PERCENTAGE,
+)
+TEMPERATURE_SENSOR_DESCRIPTION = VenstarSensorEntityDescription(
+    key="temperature",
+    device_class=DEVICE_CLASS_TEMPERATURE,
+    state_class=STATE_CLASS_MEASUREMENT,
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities) -> None:
+    """Set up Vensar device binary_sensors based on a config entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    hum_sensors = []
+    temp_sensors = []
+
+    sensors = coordinator.client.get_sensor_list()
+    if not sensors:
+        return
+
+    # first, look for humidity
+    for sensor in sensors:
+        if coordinator.client.get_sensor(sensor, "hum") is not None:
+            hum_sensors.append(
+                VenstarHumiditySensor(
+                    coordinator,
+                    config_entry,
+                    HUMIDITY_SENSOR_DESCRIPTION,
+                    sensor,
+                )
+            )
+
+    # Now, temperature
+    for sensor in sensors:
+        if coordinator.client.get_sensor(sensor, "temp") is not None:
+            temp_sensors.append(
+                VenstarTemperatureSensor(
+                    coordinator,
+                    config_entry,
+                    TEMPERATURE_SENSOR_DESCRIPTION,
+                    sensor,
+                )
+            )
+
+    async_add_entities(hum_sensors, True)
+    async_add_entities(temp_sensors, True)
+
+
+class VenstarSensor(VenstarEntity, SensorEntity):
+    """Base class for a Venstar sensor."""
+
+    entity_description: VenstarSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: VenstarDataUpdateCoordinator,
+        config: ConfigEntry,
+        entity_description: VenstarSensorEntityDescription,
+        sensor_name: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, config)
+        self.entity_description = entity_description
+        self.sensor_name = sensor_name
```

Comment:
You could store `name` in the `entity_description` instead

Its already supported https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L193

Reply:
I could, but I don't know the names of the sensors yet.  I only know them from calling coordinator.client.get_sensor_list(), which, I'm doing in the loop.  I guess I could build the descriptions on the fly there too, but that seems overly complex to me for some reason?

Reply:
Maybe rename the loop variable above from `sensor` to `sensor_name` as it would make the code easier to follow since the var won't have a name change between calls